homeassistant/components/smart_meter_b_route/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Smart Meter B Route integration."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE, CONF_ID, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+from .coordinator import BRouteUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type BRouteConfigEntry = ConfigEntry[BRouteUpdateCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: BRouteConfigEntry) -> bool:
+    """Set up Smart Meter B Route from a config entry."""
+
+    device = entry.data[CONF_DEVICE]
+    bid = entry.data[CONF_ID]
+    password = entry.data[CONF_PASSWORD]
+    hass.data.setdefault(DOMAIN, {})
+    coordinator = BRouteUpdateCoordinator(
+        hass, device, bid, password, update_interval=DEFAULT_SCAN_INTERVAL
+    )
+    hass.data[DOMAIN][entry.entry_id] = coordinator
```

Comment:
Instead of storing the coordinator in the hass.data, store them in `entry.runtime_data`